packages/pluggableWidgets/rich-text-web/src/utils/ckeditorPresets.ts

```diff
@@ -0,0 +1,156 @@
+    | "editingGroup"
+    | "clipboardGroup"
+    | "basicStylesGroup"
+    | "paragraphGroup"
+    | "linksGroup"
+    | "colorsGroup"
+    | "toolsGroup"
+    | "othersGroup"
+    | "stylesGroup";
+
+const TOOLBAR_GROUP: ToolbarGroup[] = [
+    { name: "document", groups: ["mode", "document", "doctools"] },
+    { name: "clipboard", groups: ["clipboard", "undo"] },
+    { name: "editing", groups: ["find", "selection", "spellchecker", "editing"] },
+    { name: "forms", groups: ["forms"] },
+    { name: "basicstyles", groups: ["basicstyles", "cleanup"] },
+    { name: "paragraph", groups: ["list", "indent", "blocks", "align", "bidi", "paragraph"] },
+    { name: "links", groups: ["links"] },
+    { name: "insert", groups: ["insert"] },
+    { name: "styles", groups: ["styles"] },
+    { name: "colors", groups: ["colors"] },
+    { name: "tools", groups: ["tools"] },
+    { name: "others", groups: ["others"] },
+    { name: "about", groups: ["about"] }
+];
+
+const TOOLBAR_ITEMS: ToolbarItems[] = [
+    {
+        name: "document",
+        items: ["Source", "-", "Save", "NewPage" /* "ExportPdf"*/, "Preview", "Print", "-", "Templates"]
+    },
+    { name: "clipboard", items: ["Cut", "Copy", "Paste", "PasteText", "PasteFromWord", "-", "Undo", "Redo"] },
+    { name: "editing", items: ["Find", "Replace", "-", "SelectAll", "-", "Scayt"] },
+    {
+        name: "forms",
+        items: ["Form", "Checkbox", "Radio", "TextField", "Textarea", "Select", "Button", "ImageButton", "HiddenField"]
+    },
+    {
+        name: "basicstyles",
+        items: [
+            "Bold",
+            "Italic",
+            "Underline",
+            "Strike",
+            "Subscript",
+            "Superscript",
+            "-",
+            "CopyFormatting",
+            "RemoveFormat"
+        ]
+    },
+    {
+        name: "paragraph",
+        items: [
+            "NumberedList",
+            "BulletedList",
+            "-",
+            "Outdent",
+            "Indent",
+            "-",
+            "Blockquote",
+            "CreateDiv",
+            "-",
+            "JustifyLeft",
+            "JustifyCenter",
+            "JustifyRight",
+            "JustifyBlock",
+            "-",
+            "BidiLtr",
+            "BidiRtl",
+            "Language"
+        ]
+    },
+    { name: "links", items: ["Link", "Unlink", "Anchor"] },
+    { name: "insert", items: ["Image", "Table", "HorizontalRule", "Smiley", "SpecialChar", "PageBreak", "Iframe"] },
+    { name: "styles", items: ["Styles", "Format", "Font", "FontSize"] },
+    { name: "colors", items: ["TextColor", "BGColor"] },
+    { name: "tools", items: ["Maximize", "ShowBlocks"] },
+    { name: "about", items: ["About"] }
+];
+
+function SET_PRESET(type: "basic" | "standard" | "full"): CKEditorConfig {
+    const config: CKEditorConfig = {};
+    let toolbarGroup: Array<ToolbarGroup | string> = [];
+    switch (type) {
+        case "basic":
+            config.toolbarGroups = [...TOOLBAR_GROUP];
```

Comment:
same as in `...config` . I want to keep original object untouched

Reply:
as mentioned before, you can just use `config`

Reply:
Same comment about `config` as others

Reply:
You cant, component renders every time when some props changed.
it means every time it will try to fetch correct preset with already modified objects.

Not for the case of "basic", because we don't modify anything here, but for the other ones, because we push "/" in specific indexes

Reply:
Can you explain why "you can't"? Based on your comment that it re-renders everytime a prop changes, your current code would mean it always re-renders because everytime this code is ran, toolbarGroups will become a new reference because you're spreading into a new array, which would be a "change".

Reply:
I don't follow you. the second part of your answer is confusing me, but i'll give a short code example to make it clear why I am spreading.

Example 1. Result without spreading. Main object modifies every time when you do some actions
```
const initialArray = [];

const result = () => {
  const modifiedArray = initialArray;
  modifiedArray.push("value");
  return modifiedArray;
}

const result1 = result();
const result2 = result();

console.log(initialArray, result1, result2);
['value', 'value'], ['value', 'value'], ['value', 'value']
```

So as you can see in our case it will push `"/"` every time when react component renders.

Example 2. When spreading initialObject it creates new object without reference so you can push your changes to the new array and always have your desired result.

```
const initialArray = [];

const result = () => {
  const modifiedArray = [...initialArray];
  modifiedArray.push("value");
  return modifiedArray;
}

const result1 = result();
const result2 = result();

console.log(initialArray, result1, result2);

[] ['value'] ['value']
```